gestor.sh

```diff
@@ -0,0 +1,337 @@
+##
+crear_asignatura() {
+
+	# Procesamos los argumentos
+	# Si no hay abreviatura, usamos el nombre SIN los espacios
+	local name="$1"
+	local abbr="${2:-${name//[[:space:]]/}}"
+
+	# Comprobamos que se haya proporcionado un nombre
+	[[ -z "$name" ]] && abort "usage: $0 [name] [abbreviation]"
+
+	# Sanity check: que no exista ya el directorio!
+	[[ -e "$name" ]] && abort "Ya existe un fichero con el nombre $name!"
+
+	INFO "Creando $name/ (abbrev: $abbr)"
+	mkdir "$name" && cd "$name"
+
+	local extra_tex_files_input=""
+
+	# Creamos algunos directorios extra
+	for subdir in img pdf tex tikzgen; do
+		INFO "Creado directorio $name/$subdir (con fichero .keep)"
+		mkdir "$subdir"
+		touch "$subdir/.keep"
+	done
+
+	# Creamos sub-documentos por defecto
+	for suffix in Ejs; do
+		texfile="tex/${abbr}_${suffix}.tex"
+		printf -v extra_tex_files_input "$extra_tex_files_input\n\chapter{---}\n\\input{$texfile}"
+		echo "% -*- root: ../${abbr}.tex -*-" > "$texfile"
+		INFO "Creado fichero extra $texfile"
+	done
+
+	local current_year=$(date +%y)
+	local docdate
+
+	# ¿Primer o segundo cuatrimestre?
+	if [[ $(date +%m ) -ge 8 ]]; then
+		docdate="$current_year/$((current_year + 1)) C1"
+	else
+		docdate="$((current_year - 1))/$current_year C2"
+	fi
+
+	# Creamos el fichero base a partir de una plantilla
+	cat > "$abbr.tex" <<-EOF
+		\documentclass[palatino]{apuntes}
+
+		\title{$name}
+		\author{}
+		\date{$docdate}
+
+		% Paquetes adicionales
+
+		% --------------------
+
+		\begin{document}
+		\pagestyle{plain}
+		\maketitle
+
+		\tableofcontents
+		\newpage
+		% Contenido.
+
+		%% Apéndices (ejercicios, exámenes)
+		\appendix
+		$extra_tex_files_input
+		\printindex
+		\end{document}
+	EOF
+
+
+	INFO "Terminado con exito!"
+}
+
+##
+## compilar
+##
+## Compila los apuntes de la asignatura especificada.
+## Uso: compilar [asignatura]
+##
+compilar() {
+
+	# Esta separado para no ignorar el codigo de returno
+	local asignatura;
+	asignatura="$(seleccionar_asignatura "$1")"
+
+	INFO "Compilando ${F_UNDER}$asignatura${F_RESET} ..."
+
+	# Preparamos para compilar...
+	cd "$asignatura"
+	mkdir -p tikzgen
+
+	# Compilamos comprobando el codigo de returno
+	if ! latexmk -r "../latexmkrc.pl" > /dev/null; then
+
+		ERR "Ha ocurrido un error fatal al compilar"
+		ERR "Informacion de diagnostico en"
+		ERR "    ${F_UNDER}$asignatura/$(echo pdf/*.log)${F_RESET}"
+		ERR "Resuelva los errores, y vuelva a intentarlo"
+		return 1
+
+	fi
+
+	INFO "Compilado con exito!"
+	return 0
+}
+
+##
+## compilar_todo
+##
+## Compila todos los apuntes.
+##
+compilar_todo() {
+
+	local dir_num=0
+	local dir_upd=0
+	local dir_err=0
+
+	local start_time=$SECONDS
+
+	while IFS= read -r -d $'\0' asignatura; do
+
+		if ( compilar "$(dirname "$asignatura")" ); then
+			(( dir_upd += 1 ))
+		else
+			(( dir_err += 1 ))
+		fi
+
+		((dir_num += 1))
+
+		INFO "================================================================"
+
+	done < <(find . -iname '*.tex' -depth 2 -print0)
```

Comment:
¿Qué version de coreutils usas?
Con coreutils 4.4.2 da el siguiente error, que hace que no encuentre nada:

    find: warning: you have specified the -depth option after a non-option argument -iname,
    but options are not positional (-depth affects tests specified before it as well as those specified after it).
    Please specify options before other arguments.

    find: paths must precede expression: 2
    Usage: find [-H] [-L] [-P] [-Olevel] [-D help|tree|search|stat|rates|opt|exec] [path...] [expression]

Creo que el problema es una opción incorrecta. El comando correcto que creo que querías poner es

     find . -iname '*.tex' -maxdepth 2 -print0